### DISCUSSION TITLE
Add support for bookmarks to specific revisions of files (magit-blob-mode)

### DISCUSSION THREAD

**Discussion by sheijk** [magit/magit#4854] [Ideas]
- Open a git repository in magit
- `l l` to run magit-log-current
- select a commit
- select a file, press `C-m` to open the old revision
- (I'm now seeing something like file-name.txt~3ca245 in magit-blob-mode)
- `C-x r m` to invoke bookmark-set

I'm getting an error message "bookmark-buffer-file-name: Buffer not visiting a file or directory". It would be nice if the buffer could be saved as bookmark to save links to specific revisions of files.

    M-x magit-version RET

Magit 20221208.1848 [>= 3.3.0.50-git], Git 2.39.0, Emacs 28.2, darwin
Also not implemented in newest git revision

When looking at magit-bookmark.el I see that there is support for several different magit modes but not for magit-blob-mode.

I've tried to add an implementation:
```elisp
(put 'magit-blob-mode 'magit-bookmark-variables
     '(magit-buffer-revision
       magit-buffer-file-name
       magit-buffer-refname))

(cl-defmethod magit-bookmark-name (&context (minor-mode magit-blob-mode))
  (format "magit-blob(%s~%s)" magit-buffer-file-name magit-buffer-revision))
```
But it fails as minor-mode is not a known specializer. Also magit-bookmark-variables seems to only work with major modes, as well

**Comment by tarsius** [comment]
Implemented in 639ea6c61598f196c363c9299d21a7301b7d9fa0. Sorry this took so long.
